app/js/views/app_view.js

```diff
@@ -1,16 +1,43 @@
 define([
   'jquery',
   'backbone',
-  'text!templates/app_view.ejs'
-], function ($, Backbone, indexTemplate) {
+  'text!templates/app_view.ejs',
+  'collections/todo'
+], function ($, Backbone, indexTemplate, todoCollection) {
   return Backbone.View.extend({
     el: $('#todo-app'),
+    KEY_ENTER_CODE: 13,
 
-    currentDate: function () {
-      return new Date();
+    events: {
+      'click #addButton': 'addItemsHandler',
+      'keydown #input': 'keyAction'
     },
 
+    addItemsHandler:function(){
+      var input = $('#input');
+      var newItem = input.value;
+      addItems(newItem);
+      input.value = "";
+    },
+
+    addItems: function(item){
+      if(_.trim(item) !== ''){
+        this.collection.add({name: item, done: false});
+      }
+    },
+
+    keyAction: function(e){
+      if(e.which === KEY_ENTER_CODE){
+        var input = $('#input');
+        var newItem = input.value;
+        addItems(newItem);
+        input.value = "";
+      }
+    },
+
+
     render: function () {
+      this.collection = new todoCollection();
```

Comment:
Not sure this is right, each time you render this view your collection is created?

What is the proper place to put this in?